File: Strivers A2Z DSA course/2. Learn Important Sorting Techniques/2.1. Sorting-I/selection-sort.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
  
    int n;
    cout << "Enter array size: ";
    cin >> n;
    cout << "Enter array: ";
    vector<int>arr(n);
    for(int i=0; i<n; i++){
        cin >> arr[i];
    }

    for(int i=0; i<n-1; i++){
        
        int mini = i;

        for(int j=i; j<n; j++){
            if(arr[j] < arr[mini]){
                mini = j;
            }
        }
        swap(arr[i], arr[mini]);
    }
    cout << "Sorted array: ";
    for(auto x: arr){
        cout << x << ", ";
    }
    cout << "\n";
    return 0;
}

File: Strivers A2Z DSA course/2. Learn Important Sorting Techniques/2.1. Sorting-I/SelectionSort.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class SelectionSort {

    public static void selectionSort(List<Integer> arr, int n) {
        for (int i = 0; i < n - 1; i++) {
            int mini = i;
            for (int j = i; j < n; j++) {
                if (arr.get(j) < arr.get(mini)) {
                    mini = j;
                }
            }
            Collections.swap(arr, i, mini);
        }
    }

    public static void main(String[] args) {

        int n;
        System.out.print("Enter the size of array: ");
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        System.out.print("Enter elements: ");
        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            arr.add(sc.nextInt());
        }
        selectionSort(arr, n);
        System.out.println("Sorted array: " + arr);
    }

}
